core/vnl/vnl_matlab_print_scalar.cxx

```diff
@@ -4,88 +4,131 @@
 #include <cstdlib>
 #include <cstring>
 #include <complex>
+
 #include "vnl_matlab_print_scalar.h"
+#include "vcl_compiler.h"
+
 
 void
-vnl_matlab_print_scalar(int v, char * buf, vnl_matlab_print_format)
+vnl_matlab_print_scalar(int v, char * buf, size_t buf_len, vnl_matlab_print_format format)
```

Comment:
You adding the parameter name `format` here might introduce a compiler warning about "format is unused". A common way to suppress such warnings is to remove the parameter name, which is probably why it was omitted. Not sure why you added it, but probably should remove it.  Not just on this line, but elsewhere below too.